spark/src/test/scala/io/delta/sharing/spark/RemoteDeltaLogSuite.scala

```diff
@@ -505,4 +491,34 @@ class RemoteDeltaLogSuite extends SparkFunSuite with SharedSparkSession {
       assert(TestDeltaSharingClient.limits === Seq(2L, 3L))
     }
   }
+
+  test("RemoteDeltaLog Initialized with metadata") {
+    val path = new Path("profileFile")
+    val table = Table(share = "share", schema = "schema", name = "table")
+    val client = new TestDeltaSharingClient()
+    client.clear()
+
+    def checkGetMetadataCalledOnce(versionAsOf: Option[Long] = None, nullable: Boolean): Unit = {
+      var deltaTableMetadata = client.getMetadata(table, versionAsOf, None)
+      assert(TestDeltaSharingClient.numMetadataCalled == 1)
```

Comment:
Could we test that even if create the log without deltaTableMetadata specified, getMetadata will be only called once due to caching?

Reply:
tested.